pkg/controller/direct/sql/sqlinstance_controller.go

```diff
@@ -138,14 +157,67 @@ func (a *sqlInstanceAdapter) Create(ctx context.Context, u *unstructured.Unstruc
 		return fmt.Errorf("resourceID is empty")
 	}
 
-	desiredGCP, err := SQLInstanceKRMToGCP(a.desired, a.refs)
+	if a.desired.Spec.CloneSource != nil {
+		return a.cloneInstance(ctx, u, log)
+	} else {
+		return a.insertInstance(ctx, u, log)
+	}
+}
+
+func (a *sqlInstanceAdapter) cloneInstance(ctx context.Context, u *unstructured.Unstructured, log klog.Logger) error {
+	desiredGCP, err := SQLInstanceKRMToGCPCloneRequest(a.desired, a.refs)
+	if err != nil {
+		return err
+	}
+
+	op, err := a.sqlInstancesClient.Clone(a.projectID, a.refs.sourceSQLInstance, desiredGCP).Context(ctx).Do()
+	if err != nil {
+		return fmt.Errorf("cloning SQLInstance %s failed: %w", a.desired.Name, err)
+	}
+
+	pollingBackoff := gax.Backoff{
+		Initial:    time.Second,
+		Max:        time.Minute,
+		Multiplier: 2,
+	}
+	for {
+		log.V(2).Info("polling", "op", op)
+
+		if op.Status == "DONE" {
+			break
+		}
+		if err := gax.Sleep(ctx, pollingBackoff.Pause()); err != nil {
+			return fmt.Errorf("waiting for SQLInstance %s clone failed: %w", a.desired.Name, err)
+		}
+		op, err = a.sqlOperationsClient.Get(a.projectID, op.Name).Do()
```

Comment:
If you have ideas or a PoC for surfacing information during this reconciliation ... that is so hot right now!

Reply:
Hmm.. probably out of scope for this PR, but am def planning to improve the code for this soon.